samples/nrf9160/secure_services/src/main.c

```diff
@@ -12,7 +12,7 @@
 #include <pm_config.h>
 #include <fw_info.h>
 
-#if defined(CONFIG_BOOTLOADER_MCUBOOT) || defined(CONFIG_SPM_SERVICE_RNG)
+#if defined(CONFIG_BOOTLOADER_MCUBOOT) || defined(CONFIG_SAMPLE_SERVICE_RNG)
```

Comment:
Should we have a overlay file which enabled the two new kconfig options you define? Both to get them compiled under CI and help those that want to test it.

Reply:
We want something like

    CONFIG_BOOTLOADER_MCUBOOT=y
    CONFIG_SECURE_BOOT=y
    CONFIG_SAMPLE_SERVICE_PREVALIDATE=y
    spm_CONFIG_SPM_SERVICE_PREVALIDATE=y
    mcuboot_CONFIG_EXT_API_PROVIDE_EXT_API_ENABLED=y

in the conf file but using prefixes like this is invalid, do you know how to set configs in spm and mcuboot?

Reply:
You can set command line arguments in testcase.yml AFAIK, then you should be able to provide overlay files to spm by using the `spm` prefix, `-Dspm_OVERLAY_CONF=something.conf` not sure about the exact name of the define to use.

Reply:
This didn't really work out for me, I tried -Dspm_CONF_FILE and -Dspm_OVERLAY_CONFIG. spm_CONF_FILE has to have a path from the SPM's CMakeLists, and spm_OVERLAY_CONFIG seems to be applied to the app, not the SPM. @tejlmand is this possible to achieve from the app's folder?

Reply:
We can try to figure this out later.

Reply:
Overlay files to child images does work.
https://github.com/nrfconnect/sdk-nrf/pull/2630#issuecomment-681021770

But overlay files should be given with absolute paths:
https://github.com/zephyrproject-rtos/zephyr/issues/23825#issuecomment-656005730

`CONF_FILE` does support relative path, but that solution has not been thought through fully, as it has some flaws that you can see based on the linked issue.

Reply:
You can also specify config options directly, without an overlay, like `cmake -DCONFIG_SAMPLE_SERVICE_PREVALIDATE=y ...` or `cmake -Dspm_CONFIG_SAMPLE_SERVICE_PREVALIDATE=y ...`.

Note that: `SAMPLE_SERVICE_RNG` defaults to `y`

Reply:
I tried specifying them directly via extra_args

    extra_args: -DCONFIG_BOOTLOADER_MCUBOOT=y -DCONFIG_SECURE_BOOT=y -DCONFIG_SAMPLE_SERVICE_PREVALIDATE=y -Dspm_CONFIG_SPM_SERVICE_PREVALIDATE=y -Dmcuboot_CONFIG_EXT_API_PROVIDE_EXT_API_ENABLED=y

But I get this warning (when running through sanitycheck, not when doing it manually with west build)

    CMake Warning:
      Manually-specified variables were not used by the project:
    
        -DCONFIG_BOOTLOADER_MCUBOOT
        -DCONFIG_SAMPLE_SERVICE_PREVALIDATE
        -DCONFIG_SECURE_BOOT
        -Dmcuboot_CONFIG_EXT_API_PROVIDE_EXT_API_ENABLED
        -Dspm_CONFIG_SPM_SERVICE_PREVALIDATE

Reply:
> But I get this warning (when running through sanitycheck, not when doing it manually with west build)

that worries me a bit.
sanitycheck should not get other warnings/errors than what is seen on command line.
How did you invoke `sanitycheck` ?

Reply:
> How did you invoke `sanitycheck` ?

Try:

    ../zephyr/scripts/sanitycheck -v -N -T samples/nrf9160/secure_services/ --platform nrf9160dk_nrf9160ns
    
    cat sanity-out/nrf9160dk_nrf9160ns/samples.nrf9160.secure_services.bootloaders/build.log | grep Manually-specified

Reply:
@tejlmand is there anything we can do with this?

Reply:
Finally figured it out, silly mistake. `extra_args` shouldn't have the `-D` prepended.